server/src/main/java/org/elasticsearch/index/store/FsDirectoryFactory.java

```diff
@@ -36,6 +37,8 @@
 
 public class FsDirectoryFactory implements IndexStorePlugin.DirectoryFactory {
 
+    private static final FeatureFlag MADV_RANDOM_FEATURE_FLAG = new FeatureFlag("madv_random");
```

Comment:
It seems ok to keep the feature flag, since it could be enabled in non-vector scenarios. But I'm also ok to either just remove it or replace it with a system property, if we wanna retain the ability to enable this but not even have it on-by-default in snapshot builds.

Reply:
@ChrisHegarty if we wish to tranform it into a regular system property, we can do that in a separate PR. However, I am not sure we want to expose it at all in v8.x as its incredibly trappy.